aten/src/ATen/cpu/vml.h

```diff
@@ -102,15 +97,27 @@ IMPLEMENT_VML(lgamma)
 
 #if AT_MKL_ENABLED() && !defined(__APPLE__)
 
+class MklThreadGuard {
+  int old_threads_;
+public:
+  MklThreadGuard(int num_threads):
+    old_threads_(mkl_set_num_threads_local(num_threads)) {
+  }
+  ~MklThreadGuard(){
+    mkl_set_num_threads_local(old_threads_);
+  }
+};
+
 // NB: LP64 MKL is the most commonly used and thus we assume it here. That means
 // we need to expect MKL_INT to be of type int, which implies int32_t in most
 // cases.
 static_assert(
     std::is_same<MKL_INT, int32_t>::value,
     "MKL_INT is assumed to be int32_t");
-#define IMPLEMENT_VML_MKL_STUB(op, mklop, type, mkltype)                    \
+#define IMPLEMENT_VML_MKL_STUB(op, mklop, type, mkltype)                \
   template <>                                                           \
-  inline void v##op(type * out, const type * in, int64_t size) {          \
+  inline void v##op(type * out, const type * in, int64_t size) {        \
+    MklThreadGuard guard(1);                                            \
```

Comment:
Did you find it necessary? I suppose MKL already checks whether it is inside the omp region or not?